### PR TITLE
[CDAP-19341] Validate URI while adding a tethering connection

### PR DESCRIPTION
**Pull request by rmstar**

### CODE REVIEW

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringClientHandler.java [hunk 1]
```diff
@@ -81,4 +84,15 @@ private void validateNamespaces(List<NamespaceAllocation> namespaceAllocations)
...
   }
+
+  private void validateEndpoint(String endpoint) throws BadRequestException {
+    if (endpoint == null) {
+      throw new BadRequestException("Endpoint must be specified");
+    }
+    try {
+      new URL(endpoint).toURI();
+    } catch (Exception e) {
+      throw new BadRequestException(String.format("Endpoint %s is not a valid URI", endpoint));
```
Comment (albertshau) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringClientHandler.java]: would be good to include the exception message to give some idea of what is wrong `URI: %s", e.getMessage()`. 


Also include e as the cause

**Review by albertshau — APPROVED** [review]

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringClientHandler.java (hunk 1 above)
Comment (rmstar) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringClientHandler.java]: fixed

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringClientHandler.java [hunk 2]
```diff
@@ -81,4 +84,15 @@ private void validateNamespaces(List<NamespaceAllocation> namespaceAllocations)
       }
     }
   }
+
+  private void validateEndpoint(String endpoint) throws BadRequestException {
+    if (endpoint == null) {
```
Comment (CuriousVini) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringClientHandler.java]: Good to check null and empty string

**Review by CuriousVini — APPROVED** [review]
just one minor comment. LGTM

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringClientHandler.java (hunk 2 above)
Comment (rmstar) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringClientHandler.java]: done

**Pull request merged**